Update documentation of `filestream` input with the new improvements (#25303)
elastic/beats: filebeat/docs/inputs/input-filestream-file-options.asciidoc

See <<regexp-support>> for a list of supported regexp patterns.

===== `prospector.scanner.include_files`

A list of regular expressions to match the files that you want {beatname_uc} to
include. By default no files are excluded. This option is the counterpart of
`prospector.scanner.exclude_files`.

> The following example configures {beatname_uc} to include files under `/var/log`:

Reviewer: Its kind of an allow list. Maybe we can make this a little more clear? E.g. by on the example we could say: "The following example configures {beatname_uc} to ignore any files that are not store in /var/log`... or "allows users to collect files from /var/log only"...

["source","yaml",subs="attributes"]
----
{beatname_lc}.inputs:
- type: {type}
  ...

>   prospector.scanner.include_files: ['/var/log/.*']

Reviewer: Regexes on absolute paths should always start with `^`.

See <<regexp-support>> for a list of supported regexp patterns.

===== `prospector.scanner.include_files`

A list of regular expressions to match the files that you want {beatname_uc} to
include. If a list of regexes is provided, only the files that are allowed by
the patterns are harvested.

By default no files are excluded. This option is the counterpart of
`prospector.scanner.exclude_files`.

> The following example configures {beatname_uc} to exlude files that

Reviewer 2: ```suggestion
The following example configures {beatname_uc} to exclude files that
```

harvester will first finish reading the file and close it after
`close.on_state_change.inactive` is reached. Then, after that, the file will be ignored.

[float]
[id="{beatname_lc}-input-{type}-ignore-inactive"]
===== `ignore_inactive`

If this option is enabled, {beatname_uc} ignores every file that has not been
updated since the selected time. Possible options are `since_first_start` and
`since_last_start`. The first option ignores every files that has not been updated since
the first start of {beatname_uc}. It is useful when the Beat might be restarted
due to configuration changes or due to a failure. The second option is the same
as the `tail_files` option of `log` input, it reads from files that has been updated

> since the start of {beatname_uc}.

Reviewer: > The second option is the same as the `tail_files` option of `log` input, it reads from files that has been updated since the start of {beatname_uc}.

Although true, we should not mention `tail_files` or `log` input. Lets keep the documentation here self contained.

harvester will first finish reading the file and close it after
`close.on_state_change.inactive` is reached. Then, after that, the file will be ignored.

[float]
[id="{beatname_lc}-input-{type}-ignore-inactive"]
===== `ignore_inactive`

If this option is enabled, {beatname_uc} ignores every file that has not been
updated since the selected time. Possible options are `since_first_start` and

> `since_last_start`. The first option ignores every files that has not been updated since

Reviewer 2: ```suggestion
`since_last_start`. The first option ignores every file that has not been updated since
```

the first start of {beatname_uc}. It is useful when the Beat might be restarted

> due to configuration changes or due to a failure. The second option tells

Reviewer 2: ```suggestion
due to configuration changes or a failure. The second option tells
```

> the Beat to read from files that has been updated since the start.

Reviewer 2: ```suggestion
the Beat to read from files that have been updated since its start.
```

The files affected by this setting fall into two categories:

* Files that were never harvested
* Files that were harvested but weren't updated since `ignore_inactive`.

> For files which were never seen before, the offset state is set to the end of

Reviewer 2: ```suggestion
For files that were never seen before, the offset state is set to the end of
```

> the file. If a state already exist, the offset is not changed. In case a file is

Reviewer 2: ```suggestion
the file. If a state already exists, the offset is not changed. If a file is
```

updated again later, reading continues at the set offset position.

> The setting relies on the modification time of the file to

Reviewer 2: ?

```suggestion
This setting relies on the modification time of the file to
```
